### PR TITLE
Add an additional CI system

### PR DESCRIPTION
**Pull request by Luni-4**
In order to test `rust-code-analysis` on different operating systems, I've added `GitHub Actions`.

Thanks in advance for your review! :)

### CODE REVIEW

**Comment by marco-c** [comment]
Unfortunately I think we can't use GitHub Actions in the mozilla GitHub org for the time being (https://bugzilla.mozilla.org/show_bug.cgi?id=1594423).
We can probably support testing on other operating systems via Taskcluster, I will figure out how.

**Comment by Luni-4** [comment]
Oh, I see. If we can use `Taskcluster` to test other operating systems, it would be fine for me.

**Comment by marco-c** [comment]
It should be possible to use Windows after https://github.com/mozilla/community-tc-config/pull/238, here's an example: https://github.com/taskcluster/taskcluster-worker-runner/blob/2ebe797d89f3dd3c814ea4f881a2a5c6306653db/.taskcluster.yml#L76-L113.

**Comment by Luni-4** [comment]
> It should be possible to use Windows after [mozilla/community-tc-config#238](https://github.com/mozilla/community-tc-config/pull/238), here's an example: https://github.com/taskcluster/taskcluster-worker-runner/blob/2ebe797d89f3dd3c814ea4f881a2a5c6306653db/.taskcluster.yml#L76-L113.

Good, thanks! Is Visual Studio already installed? Because if it's not, it will be difficult to install that every time

**Comment by marco-c** [comment]
I think we have all we need, this is the setup script: https://raw.githubusercontent.com/taskcluster/generic-worker/2509a1f8589b0b536186f5fe8f043385bf67197f/worker_types/win2012r2/bootstrap.ps1. It is also used to create the image needed to build Firefox, so a lot of things are already available.

**Comment by Luni-4** [comment]
Perfect Let me know when the `TaskCluster` solution will be ready :)

**Comment by marco-c** [comment]
https://github.com/mozilla/community-tc-config/pull/238 has been merged, we should now be able to use it.

**Comment by Luni-4** [comment]
> [mozilla/community-tc-config#238](https://github.com/mozilla/community-tc-config/pull/238) has been merged, we should now be able to use it.

Great! So are you working on that @marco-c ?

File: build.rs [hunk 1]
```diff
@@ -104,8 +105,11 @@ fn build_c(files: Vec<String>, language: &str) {
             .pic(true)
             .opt_level(get_opt_level())
             .debug(get_debug())
-            .warnings(false)
-            .flag("-std=c99");
+            .warnings(false);
+
+        if triple != "x86_64-pc-windows-msvc" {
```
Comment (calixteman) [inline build.rs]: nit: you should use `flag_if_supported`

File: tree-sitter-mozcpp/src/scanner.cc [hunk 2]
```diff
@@ -1 +0,0 @@
-../../tree-sitter-cpp/src/scanner.cc
```
Comment (calixteman) [inline tree-sitter-mozcpp/src/scanner.cc]: Why don't you keep a symbolic link here ?
Removing it means that we need to manually sync this file coming from tree-sitter-cpp.

Comment (Luni-4) [reply tree-sitter-mozcpp/src/scanner.cc]: It didn't work on `Windows`, perhaps I could fix that using the `bash` shell instead of `Powershell`.

Comment (calixteman) [reply tree-sitter-mozcpp/src/scanner.cc]: good idea :) 
simple and easy (I was thinking about a more complex solution)

Comment (Luni-4) [reply tree-sitter-mozcpp/src/scanner.cc]: Thanks to you for finding my initial horrible implementation! :)

**Comment by marco-c** [comment]
> Great! So are you working on that @marco-c ?

I started a WIP PR just to show the skeleton, not sure when I'll be able to continue it though.
Feel free to take it over if you want.

I'm closing this since we can't use GitHub Actions, feel free to open a new one if you want to take #122 over.

**Comment by Luni-4** [comment]
Perfect! Thanks a lot, @marco-c, I will try to update your PR then.

**Pull request closed without merging**